### PR TITLE
Extend Graph, CompilationUnit, and schema matching to accept optional operator version number

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70396
* #68358
* #69915
* __->__ #69914

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

### CODE REVIEW

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

File: torch/csrc/jit/frontend/schema_matching.cpp [hunk 1]
```diff
@@ -616,12 +631,48 @@ Value* emitBuiltinCall(
...
+          found_upgrader = true;
+        } else {
+          if (!isOpCurrentBasedOnUpgraderEntries(
+                  version_entry->second, graph_version.value())) {
+            TORCH_INTERNAL_ASSERT(false, "Valid upgrader must be present");
+          }
+        }
+      }
+    }
+    if (!found_upgrader)
```
Comment (qihqi) [inline torch/csrc/jit/frontend/schema_matching.cpp:667]: wrap with {}

**Review by qihqi — APPROVED** [review]

File: torch/csrc/jit/frontend/schema_matching.cpp [hunk 2]
```diff
@@ -616,12 +631,48 @@ Value* emitBuiltinCall(
...
+            findUpgrader(version_entry->second, graph_version.value());
+        if (old_schema_entry.has_value()) {
+          FunctionSchema old_schema =
+              parseSchema(old_schema_entry.value().old_schema);
+          upgrader_schemas.push_back(old_schema);
+          found_upgrader = true;
+        } else {
+          if (!isOpCurrentBasedOnUpgraderEntries(
+                  version_entry->second, graph_version.value())) {
+            TORCH_INTERNAL_ASSERT(false, "Valid upgrader must be present");
```
Comment (gmagogsfm) [inline torch/csrc/jit/frontend/schema_matching.cpp:662]: Is this also about deprecated upgrader? If so, the error message can be improved. Current error message doesn't explain what happened and isn't actionable.

Besides, `torch_internal_assert` implies that something is wrong with TorchScript implementation (ie some invariant that should never be broken is broken), can we throw a normal exception here instead?

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

**Pull request closed without merging**